server/opts.go

```diff
@@ -1678,11 +1691,54 @@ func parseExportStreamOrService(v interface{}, errors, warnings *[]error) (*expo
 				*errors = append(*errors, err)
 			}
 		}
-
 	}
 	return curStream, curService, nil
 }
 
+// parseServiceLatency returns a latency config block.
+func parseServiceLatency(v interface{}) (serviceLatency, error) {
+	latency, ok := v.(map[string]interface{})
+	if !ok {
+		return serviceLatency{}, fmt.Errorf("Expected latency entry to be a map/struct, got %T", v)
+	}
+
+	var sl serviceLatency
+
+	if v, ok := latency["sample"]; ok {
+		tok, ok := v.(token)
+		if !ok {
+			return serviceLatency{}, fmt.Errorf("Expected latency token, got %T", tok)
+		}
+
+		sampleStr, ok := tok.Value().(string)
+		if !ok {
+			return serviceLatency{}, fmt.Errorf("Expected latency sample to be a string, got %T", sampleStr)
+		}
+
+		sampleStr = strings.TrimSuffix(sampleStr, "%")
+		sample, err := strconv.ParseInt(sampleStr, 10, 8)
+		if err != nil {
+			return serviceLatency{}, fmt.Errorf("Failed to parse latency sample: %v", err)
+		}
+		sl.sampling = int8(sample)
+	}
+
+	if v, ok := latency["subject"]; ok {
+		tok, ok := v.(token)
```

Comment:
@derekcollison, does `subject` have a default value? Maybe this should throw an error if this is missing.

Reply:
No default for subject.